.github/workflows/w3c-publish.yml

```
@@ -65,8 +65,8 @@ jobs:
           W3C_ECHIDNA_TOKEN_JSAPI: ${{ secrets.W3C_ECHIDNA_TOKEN_JSAPI }}
           W3C_ECHIDNA_TOKEN_WEBAPI: ${{ secrets.W3C_ECHIDNA_TOKEN_WEBAPI }}
           ECHIDNA_DRYRUN: |-
-            ${{ (github.event_name == 'workflow_dispatch' && inputs.dry-run) ||
-                !(github.event_name == 'push' && github.repository == 'WebAssembly/spec' && github.ref == 'refs/heads/main') }}
+            ${{ !((github.event_name == 'workflow_dispatch' && !inputs.dry-run) ||
+                  (github.event_name == 'push' && github.repository == 'WebAssembly/spec' && github.ref == 'refs/heads/main')) }}
```

I think this is slightly clearer if we distribute the `!` a bit:
```suggestion
            ${{ (github.event_name != 'workflow_dispatch' || inputs.dry-run) &&
                !(github.event_name == 'push' && github.repository == 'WebAssembly/spec' && github.ref == 'refs/heads/main')) }}
```

---

Maybe. It's written like this because dry-run is already "inverted", and it was easier to think of it as "publish for real if these conditions are met". Another option is to actually reflect that in the variable name in this file, and then invert it when passing through to echidna's curl request as dry-run status. The downside of that is that we'd have to do the logic inversion of "true" to "false" in Makefile shell syntax, which seems worse than doing it in the GH script syntax here.

---

I added a comment. If you still think this suggestion is better, I don't really mind switching it.